File: TheAnvil/src/main/java/com/qualityplus/anvil/base/commands/HelpCommand.java
package com.qualityplus.anvil.base.commands;

import com.qualityplus.anvil.api.box.Box;
import com.qualityplus.anvil.base.config.Messages;
import com.qualityplus.assistant.TheAssistantPlugin;
import com.qualityplus.assistant.api.commands.CommandProvider;
import com.qualityplus.assistant.api.commands.command.AssistantCommand;
import com.qualityplus.assistant.util.StringUtils;
import com.qualityplus.assistant.util.math.MathUtils;
import eu.okaeri.commons.bukkit.time.MinecraftTimeEquivalent;
import eu.okaeri.injector.annotation.Inject;
import eu.okaeri.platform.bukkit.annotation.Delayed;
import eu.okaeri.platform.core.annotation.Component;
import net.md_5.bungee.api.chat.*;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Collections;
import java.util.List;

@Component
public final class HelpCommand extends AssistantCommand {
    private @Inject Box box;

    @Override
    public boolean execute(CommandSender sender, String[] args) {
        Messages.PluginMessages msg = box.files().messages().pluginMessages;
        sendHelpCommands(sender, args, TheAssistantPlugin.getAPI().getCommandProvider(), msg.helpHeader, msg.helpMessage, msg.helpfooter, msg.nextPage, msg.previousPage, msg.helpPageHoverMessage);
        return true;
    }

    @Override
    public List<String> onTabComplete(CommandSender commandSender, org.bukkit.command.Command command, String label, String[] args) {
        return Collections.emptyList();
    }

    @Delayed(time = MinecraftTimeEquivalent.SECOND)
    public void register(@Inject Box box){
        TheAssistantPlugin.getAPI().getCommandProvider().registerCommand(this, e -> e.getCommand().setDetails(box.files().commands().helpCommand));
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/base/commands/HelpCommand.kt
package com.qualityplus.anvil.base.commands

import com.qualityplus.anvil.api.box.Box
import com.qualityplus.assistant.TheAssistantPlugin
import com.qualityplus.assistant.api.commands.command.AssistantCommand
import com.qualityplus.assistant.api.commands.setup.event.CommandSetupEvent
import eu.okaeri.commons.bukkit.time.MinecraftTimeEquivalent
import eu.okaeri.injector.annotation.Inject
import eu.okaeri.platform.bukkit.annotation.Delayed
import eu.okaeri.platform.core.annotation.Component
import org.bukkit.command.Command
import org.bukkit.command.CommandSender

@Component
class HelpCommand : AssistantCommand() {
    @Inject
    private val box: Box? = null
    override fun execute(sender: CommandSender, args: Array<String>): Boolean {
        val msg = box!!.getFiles()!!.messages()!!.pluginMessages
        sendHelpCommands(
            sender,
            args,
            TheAssistantPlugin.getAPI().commandProvider,
            msg!!.helpHeader,
            msg.helpMessage,
            msg.helpfooter,
            msg.nextPage,
            msg.previousPage,
            msg.helpPageHoverMessage
        )
        return true
    }

    override fun onTabComplete(
        commandSender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ): List<String> {
        return emptyList()
    }

    @Delayed(time = MinecraftTimeEquivalent.SECOND)
    fun register(@Inject box: Box) {
        TheAssistantPlugin.getAPI().commandProvider.registerCommand(this) { e: CommandSetupEvent<AssistantCommand> ->
            e.command.setDetails(
                box.getFiles()!!.commands()!!.helpCommand
            )
        }
    }
}